modules/tracing/mule-tracer-configuration-impl/src/test/java/org/mule/runtime/tracing/level/impl/config/AutoConfigurableTracingLevelConfigurationTestCase.java

```diff
@@ -0,0 +1,158 @@
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.tracing.level.impl.config;
+
+import static org.mule.runtime.tracing.level.api.config.TracingLevel.MONITORING;
+import static org.mule.test.allure.AllureConstants.Profiling.PROFILING;
+import static org.mule.test.allure.AllureConstants.Profiling.ProfilingServiceStory.TRACING_CONFIGURATION;
+
+import static java.lang.Boolean.TRUE;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.tracer.exporter.config.api.SpanExporterConfiguration;
+import org.mule.runtime.tracing.level.api.config.TracingLevel;
+
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+import org.junit.Test;
+
+@Feature(PROFILING)
+@Story(TRACING_CONFIGURATION)
+public class AutoConfigurableTracingLevelConfigurationTestCase {
+
+  private static final TracingLevel DEFAULT_LEVEL = MONITORING;
+  private static final String CONF_FOLDER = "conf";
+  private static final String TRACING_LEVEL_EMPTY_CONF = "tracing-level-empty.conf";
+  private static final String NON_EXISTENT_CONF = "non-existent.conf";
+  private static final String WRONG_LEVEL_CONF = "wrong-level.conf";
+
+  @Test
+  public void whenNoPropertyIsInTheFileDefaultLevelIsReturned() {
+    MuleContext muleContext = mock(MuleContext.class);
+    FileTracingLevelConfiguration fileTracingLevelConfiguration =
+        new TestEmptyFileTracingLevelConfiguration(muleContext);
+    AutoConfigurableTracingLevelConfiguration tracingLevelConfiguration =
+        new AutoConfigurableTracingLevelConfiguration(muleContext);
+    tracingLevelConfiguration.setDelegate(fileTracingLevelConfiguration);
+    tracingLevelConfiguration.setSpanExporterConfiguration(mock(SpanExporterConfiguration.class));
+    assertThat(tracingLevelConfiguration.getTracingLevel(), is(DEFAULT_LEVEL));
+  }
+
+  @Test
+  public void whenNoFileExistsDefaultLevelIsReturned() {
+    SpanExporterConfiguration spanExporterConfiguration = mock(SpanExporterConfiguration.class);
+    when(spanExporterConfiguration.getStringValue(any(), any())).thenReturn(TRUE.toString());
+    MuleContext muleContext = mock(MuleContext.class);
+    FileTracingLevelConfiguration fileTracingLevelConfiguration =
+        new TestNoFileTracingLevelConfiguration(muleContext);
+    AutoConfigurableTracingLevelConfiguration tracingLevelConfiguration =
+        new AutoConfigurableTracingLevelConfiguration(muleContext);
+    tracingLevelConfiguration.setDelegate(fileTracingLevelConfiguration);
+    tracingLevelConfiguration.setSpanExporterConfiguration(spanExporterConfiguration);
+    assertThat(tracingLevelConfiguration.getTracingLevel(), is(DEFAULT_LEVEL));
+  }
+
+  @Test
+  public void whenLevelIsWrongInFileDefaultLevelIsReturned() {
```

Comment:
This tests (and the rest) need to go back to the FIleConfigurationTestCase and test for the expected result there (error, empty response or whatever should the FileConfiguration return) Here we need to mock the delegate and test the conditions where we return the default or not.